improvement: allow values without quotes

tgodzik opened the pull request:
Previously, string values would always require quotes since that coresponded to actual scala syntax. Now, we no longer allow using directives in normal Scala code, so we can allow values without quotes. Whenever there is a more complex value like one that requires a space, we would still require them.

Related to https://github.com/VirtusLab/scala-cli/issues/1954

Additionally the values are not treated literally and can contain any special characters since we are not limited by Scala syntax.

I also changed to only allow single line, which simplified tests a bit as they were taking into account newlines in a lot of places.

tgodzik committed:
improvement: allow values without quotes

Previously, string values would always require quotes since that coresponded to actual scala syntax. Now, we no longer allow using directives in normal Scala code, so we can allow values without quotes. Whenever there is a more complex value like one that requires a space, we would still require them.

tgodzik committed:
improvement: Allow any key or value oneline

szymon-rd commented on src/main/java/com/virtuslab/using_directives/custom/Parser.java:
```diff
@@ -266,11 +255,14 @@ UsingPrimitive primitive(int keyEnd) {
-              "Expected primitive value: string, numeric or boolean but found %s. %s",
-              in.td.toTokenInfoString(), solution));
+    } else if (in.td.token == Tokens.IDENTIFIER) {
+      if (in.td.name.startsWith("-") && in.td.name.chars().allMatch(Character::isDigit)) {
```
i am not sure I understand. `-42` begins with `-`, but `-` Is not a digit so this check will not pass?

tgodzik replied:
Yeah, you are right, it doesn't work. I wonder though should we care about numbers. I changed it for now, but it's a bit hacky to make it work. Should we maybe drop numeric values? It can be parsed later on if needed.

tgodzik committed:
refactor: Remove numeric and braces

Numeric values do not need to be treated differently than string ones and especially since it's not needed to use quotes it makes it problematic to check what is a number.

I also removed any kind of braces, parens etc. since I would need to fix the tests and they no longer make sense with this version of directives.

szymon-rd commented on src/test/java/com/virtuslab/using_directives/parser/ParserNumericTest.java:
```diff
@@ -82,8 +80,9 @@ public void testHexNegativeNumeric() {
 
   @Test
   public void testFailTrailingSeparator() {
```
change the name plz

tgodzik replied:
Done!

szymon-rd commented on src/test/java/com/virtuslab/using_directives/parser/ParserStringTest.java:
```diff
@@ -26,8 +49,9 @@ public void testMultilineString() {
 
   @Test
   public void testNotAllowStringInterpolator() {
```
testNotAllow -> testAllow

tgodzik replied:
And done!
